src/input/source/iio.rs

```diff
@@ -119,15 +120,26 @@ impl IIODevice {
             return Err("Unable to determine IIO driver because no name was found".into());
         };
 
-        // IIO IMU Driver
-        if glob_match(
-            "{i2c-10EC5280*,i2c-BMI*,accel-display,bmi*-imu,gyro_3d,accel_3d}",
-            name.as_str(),
-        ) {
+        // BMI IMU Driver
+        if glob_match("{i2c-10EC5280*,i2c-BMI*,bmi*-imu}", name.as_str()) {
             log::info!("Detected IMU: {name}");
             let tx = self.composite_tx.clone();
             let rx = self.rx.take().unwrap();
-            let mut driver = iio_imu::IMU::new(
+            let mut driver = bmi_imu::IMU::new(
+                self.info.clone(),
+                self.config.clone(),
+                tx,
+                rx,
+                self.get_id(),
+            );
+            driver.run().await?;
+
+        // AccelGryo3D IMU Driver
+        } else if glob_match("{gyro_3d,accel_3d}", name.as_str()) {
```

Comment:
Maybe we can create an enum and write a small function to get the IMU type like we did for hidraw?

Reply:
Done